subprojects/core/src/main/java/org/gradle/api/internal/file/copy/DefaultCopySpec.java

```diff
@@ -71,7 +71,14 @@
     private Integer fileMode;
     private Boolean caseSensitive;
     private Boolean includeEmptyDirs;
-    private DuplicatesStrategy duplicatesStrategy = DuplicatesStrategy.INHERIT;
+
+    private static final DuplicatesStrategy DUPLICATION_STRATEGY_DEFAULT = DuplicatesStrategy.INHERIT;
+    private DuplicatesStrategy duplicatesStrategy = getDefaultDuplicatesStrategy();
```

Comment:
Can't we say in the constructor of `AbstractArchiveTask`:
```
getRootSpec().setDuplicatesStrategy(DuplicatesStrategy.FAIL);
```

And that would be everything?

Reply:
I can but this will create one strategy originally (inherit) and later override it (fail) - i.e. the intermediary state will be invalid.
Is that okay?
<img src="https://user-images.githubusercontent.com/1841944/62311239-b1537000-b48b-11e9-9c14-9a054e39b8f5.png">

Reply:
Done